CMakeLists.txt

```diff
@@ -20,6 +20,41 @@ if(NOT Poco_FOUND)
       -DPOCO_UNBUNDLED:BOOL=ON
       -DCMAKE_C_FLAGS="-Wno-shift-negative-value")
```

Comment:
That also introduces a certain amount of duplication, but whatever. I just pushed a new revision implementing your suggestion and reverting the version of CMake required.